lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+Данный алгоритм оперировал словами и находил ключевые слова для текста. Чтобы позволить этому алгоритму находить наиболее важные предложения, 
+необходимо внести несколько модификаций (модификации выделены жирным шрифтом): 
+
+* $V$ - вершина графа (то есть **предложение**)
+* $d$ - damping factor, то есть вероятность перейти от этой вершины к любой другой, его значение обычно равно `0.85`
+* $InOut(V)$ - вершины, **похожие на вершину** $V$
+* $|InOut(V)|$ - количество вершин, **похожих на** $V$ (то есть, как мы назвали это в предыдущем пункте, `InOut score`)
+* $k$ - это номер итерации: при переподсчёте веса вершины мы опираемся на веса инцидентных с ней вершин с *прошлых итераций*
+    
+Таким образом, чтобы посчитать новый вес для новой вершины, нужно:
+* перебрать все вершины, инцидентные с данной вершиной
+* для каждой из таких вершин найти ее `InOut score`, использовать его как знаменатель в дроби $\frac{1}{|InOut(V_{j})|}$ и умножить полученное значение на вес этой же вершины
+* полученные значения просуммировать и умножить на $d$
+* далее результат умножения сложить с $(1 - d)$; результат такого сложения становится новым весом рассматриваемой вершины
+
+В этом шаге Вы будете реализовывать `TextRankSummarizer`.
+
+#### Шаг 7.1 Инициализатор класса
+
+Инициализатор класса `TextRankSummarizer` принимает на вход только заполненный экземпляр класса `SimilarityMatrix`.
+
+У каждого объекта `TextRankSummarizer` **обязательно** должны быть следующие атрибуты:
+* `_graph` - здесь хранится пришедший на вход заполненная матрица, которая, как мы помним, является репрезентацией текста
+* `_damping_factor` - как упоминалось ранее, это константное значение, нужное для того, чтобы алгоритм не зациклился и отражающее вероятность перейти от конкретной вершины к любой другой; установите значение данного атрибута как 0.85
+* `_convergence_threshold` - это константное значение, отражающее максимально допустимую разницу между весами до и после обновления: если после очередной итерации пересчета весов вершин сумма разностей вершин не превышает этот порог, то мы считаем, что оптимальные веса найдены; установите значение данного атрибута как 0.0001
+* `_max_iter` - это константное значение, отражающее максимально допустимое количество итераций обновления весов: теоретически на вход может прийти очень сложный граф, в котором невозможно подобрать стабильные веса, и чтобы не попасть в бесконечный цикл, мы ограничиваем максимальное количество итераций; установите значение данного атрибута как 50
+* `_scores` - веса вершин, здесь ключами являются вершины (предложения), а значениями их вес, отражающий их важность
+
+Интерфейс:
+```py
+class TextRankSummarizer:
+    def __init__(self, graph: SimilarityMatrix) -> None:
+        pass
+```
+
+#### Шаг 7.2 Обновление веса конкретной вершины
+
+Модифицируйте обновление веса конкретной вершины. Метод принимает на вход вершину, вес которой необходимо пересчитать, 
```

Comment:
so it sounds that I was right and we provide train to students??

Reply:
No, they should get it from the previous lab.

Reply:
why??